scripts/test_schema_fallback.py

```
@@ -0,0 +1,43 @@
+            if fallback != schema_id:
+                content = open(path, 'r', encoding='utf-8', errors='ignore').read()
+                content = re.sub(r"FILE_SCHEMA\(\('(.*?)'\)\)", f"FILE_SCHEMA(('"+fallback+"'))", content)
+                return ifcopenshell.file.from_string(content)
+        raise
+
+def create_test_file(path):
+    model = ifcopenshell.file(schema='IFC4X3_ADD2')
+    model.createIfcProject('1')
+    tmp = path + '.orig'
+    model.write(tmp)
+    txt = open(tmp).read().replace('IFC4X3_ADD2', 'IFC4X3_RC1')
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)